src/azure-cli-core/azure/cli/core/application.py

```diff
@@ -85,15 +83,43 @@ def execute(self, unexpanded_argv):
         self.raise_event(self.COMMAND_PARSER_LOADED, parser=self.parser)
 
         if len(argv) == 0:
+            enable_autocomplete(self.parser)
             az_subparser = self.parser.subparsers[tuple()]
             _help.show_welcome(az_subparser)
             log_telemetry('welcome')
             return None
 
         if argv[0].lower() == 'help':
             argv[0] = '--help'
+        
+        # Rudimentary parsing to get the command
```

Comment:
When the user runs `az`, we need to have loaded the commands from all modules to show the list of available commands and command groups.

Also, as command groups can be made up of commands from multiple packages, we need to load all commands from all modules to know what's available.